mersenne-31/src/mersenne_31.rs

```diff
@@ -487,5 +487,24 @@ mod tests {
         assert_eq!(F::TWO.exp_u64(1717986917).exp_const_u64::<5>(), F::TWO);
     }
 
+    #[test]
+    fn test_from_canonical_u32_valid() {
+        // Valid values within the canonical range
+        assert_eq!(Mersenne31::from_canonical_u32(0).value, 0);
+        assert_eq!(Mersenne31::from_canonical_u32(1).value, 1);
+        assert_eq!(Mersenne31::from_canonical_u32(123456).value, 123456);
+        assert_eq!(
+            Mersenne31::from_canonical_u32(Mersenne31::ORDER_U32 - 1).value,
+            Mersenne31::ORDER_U32 - 1
+        );
+    }
+
+    #[test]
+    #[should_panic]
```

Comment:
Hmm it seems like this would fail if doing `cargo test --release`. Technically the API just says it's undefined behavior, so maybe best not to test for a panic specifically, since in a way it'd be valid to remove the current assert

Reply:
Yes ok fair point I just told myself that as the CI was launched in debug mode should be fine, feel free to close this or I can just remove this panic test and keep the other one, as you prefer :)